src/lib/core/redux/slices/localMedia.ts

```diff
@@ -74,120 +110,128 @@ export const localMediaSlice = createSlice({
-                let microphoneEnabled = false;
-
-                const audioTrack = stream.getAudioTracks()[0];
-                const videoTrack = stream.getVideoTracks()[0];
-
-                if (audioTrack) {
-                    microphoneDeviceId = audioTrack.getSettings().deviceId;
-                    microphoneEnabled = audioTrack.enabled;
-                }
+        builder.addCase(doAppJoin, (state, action) => {
+            return {
+                ...state,
+                options: action.payload.localMediaOptions,
+            };
+        });
+        builder.addCase(doSetDevice.pending, (state, action) => {
+            const { audio, video } = action.meta.arg;
+            return {
+                ...state,
+                isSettingCameraDevice: video,
+                isSettingMicrophoneDevice: audio,
+            };
+        });
+        builder.addCase(doSetDevice.fulfilled, (state, action) => {
+            const { audio, video } = action.meta.arg;
+            return {
+                ...state,
+                isSettingCameraDevice: video ? false : state.isSettingCameraDevice,
+                isSettingMicrophoneDevice: audio ? false : state.isSettingMicrophoneDevice,
+            };
+        });
+        builder.addCase(doSetDevice.rejected, (state, action) => {
+            const { audio, video } = action.meta.arg;
+            return {
+                ...state,
+                isSettingCameraDevice: video ? false : state.isSettingCameraDevice,
+                isSettingMicrophoneDevice: audio ? false : state.isSettingMicrophoneDevice,
+                cameraDeviceError: video ? action.error : state.cameraDeviceError,
+                microphoneDeviceError: audio ? action.error : state.microphoneDeviceError,
+            };
+        });
+        builder.addCase(doToggleCamera.pending, (state) => {
+            return {
+                ...state,
+                isTogglingCamera: true,
+            };
+        });
+        builder.addCase(doToggleCamera.fulfilled, (state) => {
+            return {
+                ...state,
+                isTogglingCamera: false,
+            };
+        });
+        builder.addCase(doUpdateDeviceList.fulfilled, (state, action) => {
+            return {
+                ...state,
+                devices: action.payload.devices,
+            };
+        });
+        builder.addCase(doStartLocalMedia.pending, (state) => {
+            return {
+                ...state,
+                status: "starting",
+            };
+        });
+        builder.addCase(doStartLocalMedia.fulfilled, (state, { payload: { stream, onDeviceChange } }) => {
+            let cameraDeviceId = undefined;
+            let cameraEnabled = false;
+            let microphoneDeviceId = undefined;
+            let microphoneEnabled = false;
+
+            const audioTrack = stream.getAudioTracks()[0];
+            const videoTrack = stream.getVideoTracks()[0];
+
+            if (audioTrack) {
```

Comment:
Are these two really needed (checking audio and video tracks) now we have the `localStreamMetadataUpdated`?